charts/telepresence/values.yaml

```diff
@@ -341,3 +341,6 @@ client:
 
     # Tell client's DNS resolver to always send names with these suffixes to the cluster side resolver
     includeSuffixes: []
+
+trafficManager:
+  mode: single
```

Comment:
What happens if the user does the following:
```
telepresence helm install --team-mode
```
and then, at a later time, does:
```
telepresence helm install --upgrade
```
From the looks of it, that'd make this default kick in, and thus revert the earlier setting. I would expect that unless the mode is explicitly given, the current mode would survive an upgrade.

Reply:
I think the configmap would preserve the mode

Reply:
From what I could find: helm defaults to using the `reuse-values` flag if the `--set` flag is _not_ used so in the case of a plain `telepresence helm install --upgrade` we should be fine. However i think this means that other values will be reset when using the `--X-mode` (or when any other value is changed, not just the mode) unless we add the appropriate flag.

https://medium.com/@kcatstack/understand-helm-upgrade-flags-reset-values-reuse-values-6e58ac8f127e

Reply:
Perhaps we can do it like this then.
- If `--X-mode` is combined with a `--upgrade`, and no `--values` or `--set` is provided, then we retain the default (i.e., we add the `--reuse-values` flag).
- If `--X-mode` is not set, but the user provides `--upgrade` together with a `--values` or `--set` flag, then we must somehow ensure that the current `--X-mode` is preserved.

Reply:
We should probably add the `--reuse-values` flag too, at some point. Not necessary in this PR though, unless it makes the implementation easier.

Reply:
> If --X-mode is not set, but the user provides --upgrade together with a --values or --set flag, then we must somehow ensure that the current --X-mode is preserved.

Since the mode is in the `values.yaml` I don't think we should make it special by having it be immune to being reset